docs/operators/regexparser.md

```diff
@@ -0,0 +1,266 @@
+
+The regex pattern has to match the tuple in its entirety. Valid records will be emitted as POJOs while invalid ones are emitted on the error port with an error message.
+
+**Note**: field names of POJO must match field names in schema and in the same order as it appears in the incoming data.
+
+## Overview
+The operator is **idempotent**, **fault-tolerant** and **partitionable**.
+
+## Operator Information
+1. Operator location: **_malhar-contrib_**
+2. Available since: **_3.7.0_**
+3. Operator state: **_Evolving_**
+4. Java Package: [com.datatorrent.contrib.parser.RegexParser](https://github.com/apache/apex-malhar/blob/master/contrib/src/main/java/com/datatorrent/contrib/parser/RegexParser.java)
+
+
+## <a name="props"></a>Properties of RegexParser
+User needs to set the `schema` JSON string which describes data as well as specify the format on date fields if any.
+e.g.
+
+```xml
+    <property>
+        <name>dt.application.{ApplicationName}.operator.{OperatorName}.prop.schema</name>
+        <value>{
+                 "fields": [
+                   {
+                     "name": "date",
+                     "type": "Date",
+                     "constraints": {
+                       "format": "yyyy:MM:dd:hh:mm:ss"
+                     }
+                   },
+                   {
+                     "name": "id",
+                     "type": "Integer"
+                   },
+                   {
+                     "name": "signInId",
+                     "type": "String"
+                   },
+                   {
+                     "name": "ipAddress",
+                     "type": "String"
+                   },
+                   {
+                     "name": "serviceId",
+                     "type": "Double"
+                   },
+                   {
+                     "name": "accountId",
+                     "type": "Long"
+                   },
+                   {
+                     "name": "platform",
+                     "type": "Boolean"
+                   }
+                 ]
+               }
+        </value>
+    </property>
+```
+
+User needs to set the `splitRegexPattern` property whose value is the regular expression that describes the pattern of the incoming data.
+
+Example for setting `splitRegexPattern` from properties.xml of the application.
+
+```xml
+    <property>
+        <name>dt.application.{ApplicationName}.operator.{OperatorName}.prop.splitRegexPattern</name>
+        <value>.+\[SEQ=\w+\]\s*(\d+:[\d\d:]+)\s(\d+)\s* sign-in_id=(\S+) .*ip_address=(\S+).* service_id=(\S+).*account_id=(\S+).*platform=(\S+)</value>
+    </property>
+```
```

Comment:
Along with this regex example, show a sample of input text and the output POJO so the user can see exactly how the input is transformed to output via the regex; point out which parts of the regex match the specific fields.